protocol/message_validator.go

```diff
@@ -2,17 +2,52 @@ package protocol
 
 import (
 	"errors"
+	"math/big"
 	"strconv"
 	"strings"
 
 	"github.com/status-im/status-go/protocol/protobuf"
+	"github.com/status-im/status-go/protocol/v1"
 )
 
-func ValidateReceivedPairInstallation(message *protobuf.PairInstallation) error {
-	if message.Clock == 0 {
+// maxWhisperDrift is how many milliseconds we allow the clock value to differ
+// from whisperTimestamp
+const maxWhisperDriftMs uint64 = 120000
+
+func validateClockValue(clock uint64, whisperTimestamp uint64) error {
+	if clock == 0 {
 		return errors.New("clock can't be 0")
 	}
 
+	c := new(big.Int).SetUint64(clock)
+	w := new(big.Int).SetUint64(whisperTimestamp)
+	result := new(big.Int).Sub(c, w)
```

Comment:
If we only subtract lower from higher, we should never overflow and we wouldn't need to use `big.Int`, right?